xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -63,21 +63,50 @@ viewers.Comments = Class.create({
   },
   /** Enhance the Comments UI with JS behaviors. */
   startup : function () {
-    if ($("commentform")) {
-      this.form = $("commentform").up("form");
-    } else {
-      this.form = undefined;
-    }
+    this.formToggled = false;
     this.loadIDs();
     this.addReplyListener();
-    this.addSubmitListener(this.form);
-    this.addCancelListener();
     this.addEditListener();
-    this.reloadEditor({
-      callback: function () {
-        this.addPreview(this.form);
-      }.bind(this)
-    });
+    // replaces the comment button with the form on the first click.
+
+    var openCommentForm = $("OpenCommentForm");
+    if(openCommentForm) {
+      openCommentForm.observe('click', function (event) {
+        event.stop();
+        this.toggleFormHidden()
+        this.reloadEditor({
+          callback: function () {
+            this.getForm();
+            this.addSubmitListener(this.form);
+            this.addCancelListener();
+            this.addPreview(this.form);
+          }.bind(this)
+        });
+      }.bind(this));
+    }
+  },
+  // update the form element reference
+  getForm: function () {
+    var commentform = $("commentform");
+    if (commentform) {
+      this.form = commentform.up("form");
+    } else {
+      this.form = undefined;
+    }
+  },
+  // load the form only once, when it is first needed
+  // the next calls to this method does nothing but calling the callback
+  // function parameter.
+  toggleFormHidden: function () {
```

Comment:
Vocabulary: it's not really a toggle since it remains in the same state once it's displayed. So maybe something like `displayHiddenForm` would be more appropriate?